Failing pytest with ModuleNotFoundError when using hatch integration with versioningit-onbuild option

I am using v3.1.* with Hatch v1.12.0

I have this in pyproject.toml:

```
[tool.hatch.build.hooks.versioningit-onbuild]
source-file = "src/dagcellent/__init__.py"
build-file = "dagcellent/__init__.py"
require-match = true
```

When I run my tests, it fails:
```
ImportError while importing test module '/Users/timon/work/dagcellent/dagcellent/tests/test_dag_config.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/opt/homebrew/Cellar/python@3.12/3.12.3/Frameworks/Python.framework/Versions/3.12/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_dag_config.py:7: in <module>
    from dagcellent.dynamic_dag import Config, parse_config_file
E   ModuleNotFoundError: No module named 'dagcellent.dynamic_dag'
```

When I remove the on-build part, it works again.

Any ideas if I am misconfiguring something or how to dig deeper in the problem? It seems like that my package is not in the `test` venv.

p.s. I am willing to submit a PR/provide more info.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,8 @@
-v3.2.0 (in development)
+v3.1.2 (in development)
 -----------------------
 - Drop support for Python 3.7
+- **Bugfix**: When using `versioningit` with Hatch, don't run the `onbuild`
+  step for editable builds
 
 v3.1.1 (2024-04-29)
 -------------------

--- a/docs/changelog.rst
+++ b/docs/changelog.rst
@@ -6,6 +6,8 @@ Changelog
 v3.2.0 (in development)
 -----------------------
 - Drop support for Python 3.7
+- **Bugfix**: When using `versioningit` with Hatch, don't run the `onbuild`
+  step for editable builds
 
 
 v3.1.1 (2024-04-29)

--- a/src/versioningit/__init__.py
+++ b/src/versioningit/__init__.py
@@ -43,7 +43,7 @@ Visit <https://github.com/jwodder/versioningit> or
 <https://versioningit.rtfd.io> for more information.
 """
 
-__version__ = "3.2.0.dev1"
+__version__ = "3.1.2.dev1"
 __author__ = "John Thorvald Wodder II"
 __author_email__ = "versioningit@varonathe.org"
 __license__ = "MIT"

--- a/src/versioningit/hatch.py
+++ b/src/versioningit/hatch.py
@@ -73,8 +73,11 @@ class OnbuildHook(BuildHookInterface):
         super().__init__(*args, **kwargs)
         self.__tmpdir = Path(tempfile.mkdtemp())
 
-    def initialize(self, _version: str, build_data: dict[str, Any]) -> None:
+    def initialize(self, version: str, build_data: dict[str, Any]) -> None:
         init_logging()
+        if self.target_name == "wheel" and version == "editable":
+            log.debug("Not running onbuild step for editable build")
+            return None
         version_source = self.metadata.hatch.version.source
         if not isinstance(version_source, VersioningitSource):
             raise RuntimeError(

--- a/test/test_end2end.py
+++ b/test/test_end2end.py
@@ -399,6 +399,10 @@ def test_editable_mode(cmd: list[str], tmp_path: Path) -> None:
         assert get_repo_status(srcdir) == status
         info = readcmd(sys.executable, "-m", "pip", "show", "mypackage")
         assert parse_version_from_metadata(info) == details.version
+        version_var = readcmd(
+            sys.executable, "-c", "import mypackage; print(mypackage.__version__)"
+        )
+        assert version_var == "NOT SET"
         for f in details.files:
             f.check(srcdir, "project")
     finally:
@@ -425,6 +429,10 @@ def test_editable_mode_hatch(tmp_path: Path) -> None:
         assert get_repo_status(srcdir) == status
         info = readcmd(sys.executable, "-m", "pip", "show", "mypackage")
         assert parse_version_from_metadata(info) == details.version
+        version_var = readcmd(
+            sys.executable, "-c", "import mypackage; print(mypackage.__version__)"
+        )
+        assert version_var == "NOT SET"
         for f in details.files:
             f.check(srcdir, "project")
     finally:

